lib/sidekiq/middleware/client/datadog.rb

```diff
@@ -0,0 +1,65 @@
+require 'sidekiq'
+require 'sidekiq/datadog/version'
+require 'sidekiq/datadog/tag_builder'
+require 'datadog/statsd'
+require 'socket'
+
+module Sidekiq
+  module Middleware
+    module Client
+      class Datadog
+        # Configure and install datadog instrumentation. Example:
+        #
+        #   Sidekiq.configure_client do |config|
+        #     config.client_middleware do |chain|
+        #       chain.add Sidekiq::Middleware::Client::Datadog
+        #     end
+        #   end
+        #
+        # You might want to also call `client_middleware` in your `configure_server` call,
+        # since enqueued jobs can enqueue other jobs.
+        #
+        # If you have other client middleware that can stop jobs from getting pushed,
+        # you might want to ensure this middleware is added last, to avoid reporting
+        # enqueues that later get stopped.
+        #
+        # Options:
+        # * <tt>:hostname</tt>    - the hostname used for instrumentation, defaults to system hostname, respects +INSTRUMENTATION_HOSTNAME+ env variable
+        # * <tt>:metric_name</tt> - the metric name (prefix) to use, defaults to "sidekiq.job_enqueued"
+        # * <tt>:tags</tt>        - array of custom tags, these can be plain strings or lambda blocks accepting a rack env instance
+        # * <tt>:skip_tags</tt>   - array of tag names that shouldn't be emitted
+        # * <tt>:statsd_host</tt> - the statsD host, defaults to "localhost", respects +STATSD_HOST+ env variable
+        # * <tt>:statsd_port</tt> - the statsD port, defaults to 8125, respects +STATSD_PORT+ env variable
+        # * <tt>:statsd</tt>      - custom statsd instance
+        def initialize(opts = {})
```

Comment:
instead of `opts`, it would be nice to accept the more modern keyword arguments or at least `def initialize(**opts)`

Reply:
So, I 100% agree that kwargs are better than an opts hash.
The reason I didn't do that is that I don't know what Ruby versions you are targetting. There's a minimum version before which we couldn't have `kwargs`, and this code seems written to work with those, so I didn't want to force a higher version if I didn't need to.

Making that change makes it, also, backwards incompatible (for the same reason), so it seemed a more onerous choice.

If you are happy with that implication, i'll happily change the 3 classes to take kwargs, and we can release a major version bump instead of a minor, I'm easy either way :)